utils/tests_fetcher.py

```diff
@@ -0,0 +1,387 @@
+    splits = content.split("\'\'\'")
+    # fmt: on
+    content = "".join(splits[::2])
+
+    # Remove empty lines and comments
+    lines_to_keep = []
+    for line in content.split("\n"):
+        # remove anything that is after a # sign.
+        line = re.sub("#.*$", "", line)
+        if line.isspace():
+            continue
+        lines_to_keep.append(line)
+    return "\n".join(lines_to_keep)
+
+
+def diff_is_docstring_only(repo, branching_point, filename):
+    """
+    Check if the diff is only in docstrings in a filename.
+    """
+    with checkout_commit(repo, branching_point):
+        with open(filename, "r", encoding="utf-8") as f:
+            old_content = f.read()
+
+    with open(filename, "r", encoding="utf-8") as f:
+        new_content = f.read()
+
+    old_content_clean = clean_code(old_content)
+    new_content_clean = clean_code(new_content)
+
+    return old_content_clean == new_content_clean
+
+
+def get_modified_python_files():
+    """
+    Return a list of python files that have been modified between the current head and the master branch.
+    """
+    repo = Repo(PATH_TO_TRANFORMERS)
+
+    print(f"Master is at {repo.refs.master.commit}")
+    print(f"Current head is at {repo.head.commit}")
+
+    branching_commits = repo.merge_base(repo.refs.master, repo.head)
+    for commit in branching_commits:
+        print(f"Branching commit: {commit}")
+
+    print("\n### DIFF ###\n")
+    code_diff = []
+    for commit in branching_commits:
+        for diff_obj in commit.diff(repo.head.commit):
+            # We always add new python files
+            if diff_obj.change_type == "A" and diff_obj.b_path.endswith(".py"):
+                code_diff.append(diff_obj.b_path)
+            # We check that deleted python files won't break correspondping tests.
+            elif diff_obj.change_type == "D" and diff_obj.a_path.endswith(".py"):
+                code_diff.append(diff_obj.a_path)
+            # Now for modified files
+            elif diff_obj.change_type in ["M", "R"] and diff_obj.b_path.endswith(".py"):
+                # In case of renames, we'll look at the tests using both the old and new name.
+                if diff_obj.a_path != diff_obj.b_path:
+                    code_diff.extend([diff_obj.a_path, diff_obj.b_path])
+                else:
+                    # Otherwise, we check modifications are in code and not docstrings.
+                    if diff_is_docstring_only(repo, commit, diff_obj.b_path):
+                        print(f"Ignoring diff in {diff_obj.b_path} as it only concerns docstrings or comments.")
+                    else:
+                        code_diff.append(diff_obj.a_path)
+
+    return code_diff
+
+
+def get_module_dependencies(module_fname):
+    """
+    Get the dependencies of a module.
+    """
+    with open(module_fname, "r", encoding="utf-8") as f:
+        content = f.read()
+
+    imports = re.findall(r"from\s+(\.+\S+)\s+import\s+\S+\s", content)
```

Comment:
We also have a few `import transformers` and then we won't know which dependencies are used as those would be fully qualified calls.

Reply:
Good point. They are mainly in the conversion scripts which do not have tests files, but for completeness, they should be added.